exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/stream/StreamingHttpConnection.java

```diff
@@ -219,12 +230,39 @@ private void emitRows(RowSetReader batchReader) throws IOException {
    */
   public void finish() throws IOException {
     JsonOutput gen = writer.jsonOutput();
-    gen.writeEndArray();
-    writeNewline(gen);
+    if (batchCount == 0) {
+      startHeader();
+      if (options != null &&
+          Boolean.parseBoolean(options.get(ExecConstants.ENABLE_VERBOSE_ERRORS_KEY))) {
```

Comment:
Do we want to reuse the rather verbose key here? It might be confusing as I would think, from the name, that I can also get the new behavior by setting the given config item.

In fact, I think we have a way to pass session options in a JSON query. Would it be better to simply use that mechanism here? Check if a new "drill.rest.errors.verbose" (or some such) option is set? This would let a site set the option by default, and would allow a future version of Drill make the verbose option the default.

Otherwise, if we want to add the one-off options here, maybe use short names such as "verbose". Someone might add a "schema-first" option to export the schema before the data to avoid client-side buffering.